pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -34,6 +34,8 @@
 else:
     date_strategy = st.datetimes()
 
+pytestmark = pytest.mark.usefixtures("pyarrow_skip")
```

Comment:
These tests used to pass but are now broken since you removed np.asarray from the datetime handling. Please revert that change and changes to these tests. ( see 
https://github.com/pandas-dev/pandas/pull/31817#discussion_r513163314, for more context.)

Reply:
No argument that we need the `np.asarray` to make the datetime handling pass but I'm going to do that in a separate PR once this is in. This PR is intrinsically large, I want to keep the scope minimal